core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -58,6 +58,17 @@ final class ShuffleBlockFetcherIterator(
 
   import ShuffleBlockFetcherIterator._
 
+  private[this] val enableExternalShuffleService =
+    blockManager.conf.getBoolean("spark.shuffle.service.enabled", false)
+
+  /**
+   * If this option enabled, bypass unnecessary network interaction
+   * if multiple block managers work in a single host.
+   */
+  private[this] val enableBypassNetworkAccess =
+    blockManager.conf.getBoolean("spark.shuffle.bypassNetworkAccess", false) &&
```

Comment:
I agree though, if this bypassing always enabled, code blocks to fetch remote blocks are not totally tested in LocalSparkContext. Thought?